hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/fs/FileSystem.java

```diff
@@ -222,6 +233,17 @@ public void initialize(URI name, Configuration conf) throws IOException {
   public String getScheme() {
     throw new UnsupportedOperationException("Not implemented by the " + getClass().getSimpleName() + " FileSystem implementation");
   }
+  
+  /**
+   * Return an alternative protocol scheme for the FileSystem.
+   * <p/>
+   * This implementation throws an <code>UnsupportedOperationException</code>.
```

Comment:
It does not throw any Exception? Should it throw though instead of returning null?

Reply:
I will remove the comment (stupid copy past). It should return something by default so that we only need to overwrite it when there is an actual alternative scheme.